### ISSUE TITLE
Unhandled TypeError while setting up key backup

### ISSUE THREAD

**Issue by LordKorea** [vector-im/riot-web#9901]
### Description

When setting up key backup, after entering the passphrase twice the next button does not respond (but is not grayed out). An error is displayed in the console (Ctrl+Shift+I)

### Steps to reproduce

- Have no key backup set up
- Click on "Start using Key Backup" (in Settings > Security & Privacy)
- Enter a passphrase
- Click next
- Confirm the passphrase
- Click next
- At this point, nothing happens (an error is shown on the Ctrl+Shift+I console)

#### Describe how what happens differs from what you expected.

Starting to use key backup should work.

Log: fails due to "HTTP 500"    

Stuck in this modal window:
![stuck1](https://user-images.githubusercontent.com/31963040/58633704-8876f900-82e9-11e9-9032-e4cfc3e5bd4f.png)

The error in the console:
```
Unhandled rejection TypeError: Expected Buffer
    at Object.encode (vector://vector/webapp/bundles/41e5560…/bundle.js:78:92893)
    at t.encodeRecoveryKey (vector://vector/webapp/bundles/41e5560…/bundle.js:78:91136)
    at X.<anonymous> (vector://vector/webapp/bundles/41e5560…/bundle.js:63:161320)
    at E (vector://vector/webapp/bundles/41e5560…/bundle.js:30:51859)
    at Generator._invoke (vector://vector/webapp/bundles/41e5560…/bundle.js:30:51647)
    at Generator.e.(anonymous function) [as next] (vector://vector/webapp/bundles/41e5560…/bundle.js:30:52038)
    at Generator.c (vector://vector/webapp/bundles/41e5560…/bundle.js:1:84882)
    at p._promiseFulfilled (vector://vector/webapp/bundles/41e5560…/bundle.js:1:41754)
    at F._settlePromise (vector://vector/webapp/bundles/41e5560…/bundle.js:1:58951)
    at F._settlePromise0 (vector://vector/webapp/bundles/41e5560…/bundle.js:1:59549)
    at F._settlePromises (vector://vector/webapp/bundles/41e5560…/bundle.js:1:60901)
    at f (vector://vector/webapp/bundles/41e5560…/bundle.js:1:13003)
    at p (vector://vector/webapp/bundles/41e5560…/bundle.js:1:12942)
    at l._drainQueues (vector://vector/webapp/bundles/41e5560…/bundle.js:1:14450)
    at drainQueues (vector://vector/webapp/bundles/41e5560…/bundle.js:1:12684)
From previous event:
    at F.U [as _captureStackTrace] (vector://vector/webapp/bundles/41e5560…/bundle.js:1:27777)
    at new p (vector://vector/webapp/bundles/41e5560…/bundle.js:1:40629)
    at t.<anonymous> (vector://vector/webapp/bundles/41e5560…/bundle.js:1:43718)
    at t._onPassPhraseConfirmNextClick (vector://vector/webapp/bundles/41e5560…/1.js:1:3753)
    at Object.o (vector://vector/webapp/bundles/41e5560…/bundle.js:46:130509)
    at s (vector://vector/webapp/bundles/41e5560…/bundle.js:46:128710)
    at Object.executeDispatchesInOrder (vector://vector/webapp/bundles/41e5560…/bundle.js:46:129439)
    at d (vector://vector/webapp/bundles/41e5560…/bundle.js:30:59064)
    at f (vector://vector/webapp/bundles/41e5560…/bundle.js:30:59190)
    at Array.forEach (<anonymous>)
    at e.exports (vector://vector/webapp/bundles/41e5560…/bundle.js:63:109690)
    at Object.processEventQueue (vector://vector/webapp/bundles/41e5560…/bundle.js:30:60582)
    at vector://vector/webapp/bundles/41e5560…/bundle.js:73:37497
    at Object.handleTopLevel [as _handleTopLevel] (vector://vector/webapp/bundles/41e5560…/bundle.js:73:37519)
    at f (vector://vector/webapp/bundles/41e5560…/bundle.js:73:58580)
    at u.perform (vector://vector/webapp/bundles/41e5560…/bundle.js:46:20749)
    at Object.batchedUpdates (vector://vector/webapp/bundles/41e5560…/bundle.js:63:124785)
    at Object.batchedUpdates (vector://vector/webapp/bundles/41e5560…/bundle.js:23:57501)
    at dispatchEvent (vector://vector/webapp/bundles/41e5560…/bundle.js:73:59361)
```

### Version information

- **Platform**: Desktop

For the desktop app:

- **OS**: Linux arch 5.1.5-arch1-2-ARCH
- **Version**: 1.2.0

I confirmed the problem exists for at least one other person (on windows).

**Comment by jryans** [comment]
Thanks for the report. I can confirm that https://riot.im/app is also affected. 😭 We'll work on patching this up.

**Comment by jryans** [comment]
Most likely the wrong version of `base-x` is being used and causing this issue. A quick fix is likely to copy the Yarn resolution blocks from JS SDK to React SDK and Riot as well, but that starts to feel like an annoying workaround for all JS SDK consumers...

With a little more time, we could upgrade to latest `base-x` and fix up our usage of it to match, or perhaps fork the package so we don't have to worry about it in the future.

**Comment by jryans** [comment]
We're intending to release a JS SDK 2.0.0 and Riot 1.2.1 with this fixed tomorrow. In the mean time, if you're able to use https://riot.im/develop, that version of the app has key backup setup working properly.

**Comment by jryans** [comment]
This is now fixed in Riot 1.2.1 which is live on https://riot.im/app. Electron builds are also published as well. Thanks for your patience!
